salt/search/java.sls

```diff
@@ -0,0 +1,24 @@
+# Add the oracle jvm ppa, and install java8
+# https://launchpad.net/~webupd8team/+archive/ubuntu/java
+# NOTE: Let me know if you require openjdk version.
+
+oracle-ppa:
+  pkgrepo.managed:
+    - humanname: WebUpd8 Oracle Java PPA repository
+    - name: deb http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main
+    - dist: trusty
+    - file: /etc/apt/sources.list.d/webupd8team-java.list
+    - keyid: EEA14886
+    - keyserver: keyserver.ubuntu.com
+
+oracle-license-select:
+  debconf.set:
+    - name: oracle-java8-installer
+    - data:
+        'shared/accepted-oracle-license-v1-1': {'type': 'boolean', 'value': 'true'}
+
+oracle-java8-installer:
+  pkg:
+    - installed
+    - require:
```

Comment:
best practice would be a `require` on `oracle-license-select` too. In general states are executed in the order in which they are specified, but if we transition later to running them in parallel `requires` avoid race conditions.